DeepCat.js

```diff
@@ -209,7 +208,24 @@
 		newSearchTerms = DeepCat.computeResponses( responses, newSearchTerms );
 		newSearchTerms = DeepCat.computeErrors( errors, newSearchTerms );
 
-		substituteSearchRequest( newSearchTerms.join( ' ' ) );
+		logAndFinishRequest( newSearchTerms.join( ' ' ) );
+	}
+
+	function logAndFinishRequest( newSearchTermString ) {
+		var logRequestUrl = '//tools.wmflabs.org/catgraph-jsonp/logrequestlength?searchquerylength={0}';
+		return $.ajax( {
+			url: stringFormat( logRequestUrl, newSearchTermString.length ),
+			timeout: ajaxTimeout,
+			cache: false,
+			complete: function() {
+				finishDeepCatRequest( newSearchTermString );
+			}
+		} );
+	}
+
+	function finishDeepCatRequest( newSearchTermString ) {
+		substituteSearchRequest( newSearchTermString );
+		removeAjaxThrobber();
```

Comment:
The logging is not critical for the feature. It should be irrelevant for the user. Please remove the throbber earlier the moment the user can use the UI again.